patches/ygot/ygot.patch

```diff
@@ -1627,7 +1627,24 @@ index ec52ee3..c312ec3 100644
  	mapVal, err := makeValForInsert(schema, root, keys)
  	if err != nil {
  		return nil, fmt.Errorf("failed to create map value for insert, root %T, keys %v: %v", root, keys, err)
-@@ -557,3 +747,43 @@ func getKeyValue(structVal reflect.Value, key string) (interface{}, error) {
+@@ -524,7 +714,16 @@ func unmarshalContainerWithListSchema(schema *yang.Entry, parent interface{}, va
+ 	}
+ 	// Create a container equivalent of the list, which is just the list
+ 	// with ListAttrs unset.
++	// Since we've patched a mutex into the Entry struct we need special handling
++	// when making this copy.  We don't want to copy the mutex while it is locked
++	// by another goroutine, doing so would leave newSchema's copy of the mutex
++	// also locked but it would never be unlocked making it a deadlock risk.
++	// To avoid this, we lock the original mutex, make the copy, then unlock both
++	// the original and copy mutex.
++	schema.ChildSchemaMutex.Lock()
+ 	newSchema := *schema
++	schema.ChildSchemaMutex.Unlock()
```

Comment:
please make it a common utility function to perform the copy schema or keep it as member function of the *yang.Entry

Reply:
Updated as requested